source-browser.html

```diff
@@ -182,6 +193,29 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
       }
 
       $(document).ready(() => {
+        $(window).on('scroll resize', function(){
+            const top = $(window).scrollTop();
+            const bottom = top + $(window).height();
+            TOPICS.forEach(topic => {
+              let {name} = topic;
+              const name_position = $(`#${name}`).position().top;
+              if (top < name_position && name_position < bottom) {
+                $(`#${name}-nav-link`).hide();
+              } else {
+                $(`#${name}-nav-link`).text(`${name} ${name_position > bottom ? '↓' : '↑'}`).show();
```

Comment:
Can replace the entire block with something like:

```suggestion
                $(`#${name}-nav-link`)
                  .text(`${name} ${name_position > bottom ? '↓' : '↑'}`)
                  .toggle(top >= name_position || name_position >= bottom);
```